src/components/buttons/link-button/link-button.js

```diff
@@ -1,65 +1,109 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import { Link } from 'react-router-dom';
+import { Link as ReactRouterLink } from 'react-router-dom';
 import { css } from '@emotion/core';
+import styled from '@emotion/styled';
 import vars from '../../../../materials/custom-properties';
-import filterAriaAttributes from '../../../utils/filter-aria-attributes';
-import filterDataAttributes from '../../../utils/filter-data-attributes';
+import getPassThroughProps from '../../../utils/get-pass-through-props';
 import Text from '../../typography/text';
 import Spacings from '../../spacings';
 
-const LinkButton = props => (
-  <Link
-    to={props.to}
-    css={[
-      css`
-        display: inline-flex;
-        align-items: center;
-        font-size: 1rem;
-        border: none;
-        background: none;
-        padding: 0;
-        min-height: initial;
-        cursor: pointer;
-        text-decoration: none;
-        span {
-          color: ${props.isDisabled ? vars.colorNeutral : vars.colorPrimary};
-        }
-      `,
-      props.isDisabled &&
-        css`
-          cursor: not-allowed;
-        `,
-      !props.isDisabled &&
-        css`
-          &:hover {
-            span {
-              color: ${vars.colorPrimary25};
-            }
-
-            * {
-              fill: ${vars.colorPrimary25};
-            }
-          }
-        `,
-    ]}
-    onClick={props.isDisabled ? event => event.preventDefault() : undefined}
-    data-track-component="LinkButton"
-    {...filterAriaAttributes(props)}
-    {...filterDataAttributes(props)}
-    aria-label={props.label}
-  >
-    <Spacings.Inline scale="xs" alignItems="center">
-      {Boolean(props.iconLeft) &&
-        React.cloneElement(props.iconLeft, {
-          size: 'medium',
-          theme: props.isDisabled ? 'grey' : 'green',
-        })}
-      <Text.Body isInline={true}>{props.label}</Text.Body>
-    </Spacings.Inline>
-  </Link>
+const hoverStyles = () => css`
+  &:hover,
+  &:focus,
+  &:active {
+    span {
+      color: ${vars.colorPrimary25};
+    }
+
+    svg * {
+      fill: ${vars.colorPrimary25};
+    }
+  }
+`;
+
+const createStyledComponent = component => styled(component)`
+  display: inline-flex;
+  align-items: center;
+  font-size: 1rem;
+  border: none;
+  background: none;
+  padding: 0;
+  min-height: initial;
+  text-decoration: none;
+
+  span {
+    color: ${props => (props.disabled ? vars.colorNeutral : vars.colorPrimary)};
+  }
+
+  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
+
+  ${props => !props.disabled && hoverStyles}
+`;
+
+const StyledReactRouterLink = createStyledComponent(ReactRouterLink);
+const StyledExternalLink = createStyledComponent('a');
+
+const componentProps = ['isExternal', 'to', 'label', 'isDisabled', 'iconLeft'];
```

Comment:
This is odd to me. Is this an established pattern? I thought we could do something like `Object.keys(LinkButton.propTypes)` to determine the pass through props?

Reply:
good call!